coordinator/src/main/java/org/apache/uniffle/coordinator/conf/YamlClientConfParser.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.uniffle.coordinator.conf;
+
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.yaml.snakeyaml.Yaml;
+
+import org.apache.uniffle.common.RemoteStorageInfo;
+
+/** The conf will be stored in the yaml format file. */
+public class YamlClientConfParser implements ClientConfParser {
+  private static final String RSS_CLIENT_CONF_KEY = "rssClientConf";
+  private static final String REMOTE_STORAGE_INFOS_KEY = "remoteStorageInfos";
+
+  @Override
+  public ClientConf tryParse(InputStream fileInputStream) throws Exception {
+    Yaml yaml = new Yaml();
```

Comment:
CVE-2017-18640 , We should upgrade to a more secure version.

Reply:
Upgrade to the latest version of 2.2